src/Cake.Svn/SvnAliases.Delete.cs

```diff
@@ -0,0 +1,130 @@
+        /// </returns>
+        /// <example>
+        /// <para>Delete a directory from Subversion.</para>
+        /// <code>
+        /// <![CDATA[
+        ///     var directoryRemoved = SvnDelete(@"C:\project\src\newfolder\");
+        ///
+        ///     Verbose("Directory removed: {0}", directoryRemoved);
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Delete")]
+        [CakeNamespaceImport("Cake.Svn.Delete")]
+        public static bool SvnDelete(this ICakeContext context, DirectoryPath directory)
+        {
+            return SvnDelete(context, directory, null);
+        }
+
+        /// <summary>
+        /// Delete a directory from Subversion.
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <param name="directory">The directory to delete.</param>
+        /// <param name="settings">The settings.</param>
+        /// <returns>
+        /// <c>true</c> if the command was successful. Otherwise <c>false</c> will be returned.
+        /// </returns>
+        /// <example>
+        /// <para>Delete a directory from Subversion.</para>
+        /// <code>
+        /// <![CDATA[
+        ///     var svnAddSettings = new SvnDeleteSettings
+        ///     {
+        ///         Force = true
+        ///     };
+        ///
+        ///     var directoryRemoved = SvnDelete(@"C:\project\src\newfolder\", svnAddSettings);
+        ///
+        ///     Verbose("Directory removed: {0}", directoryRemoved);
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Delete")]
+        [CakeNamespaceImport("Cake.Svn.Delete")]
+        public static bool SvnDelete(this ICakeContext context, DirectoryPath directory, SvnDeleteSettings settings)
+        {
+            var commands = new SvnDeleter(context.Environment, SvnClientFactoryMethod);
+
+            return commands.Delete(directory, settings ?? new SvnDeleteSettings());
+        }
+
+        /// <summary>
+        /// Delete a file from Subversion.
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <param name="file">The file.</param>
+        /// <returns>
+        /// <c>true</c> if the command was successful. Otherwise <c>false</c> will be returned.
+        /// </returns>
+        /// <example>
+        /// <para>Delete a file from Subversion.</para>
+        /// <code>
+        /// <![CDATA[
+        ///     var fileRemoved = SvnDelete(@"C:\project\src\newfile.cs");
+        ///
+        ///     Verbose("File removed: {0}", fileRemoved);
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Delete")]
+        [CakeNamespaceImport("Cake.Svn.Delete")]
+        public static bool SvnDelete(this ICakeContext context, FilePath file)
```

Comment:
Will be ambiguous between  `SvnDelete(this ICakeContext context, DirectoryPath directory)` if user writes `SvnDelete("foo.cake")`. Should we rename to `SvnDeleteFile`?